include/led.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Internal driver specific representation of an LED device
+ */
+
+struct led_data {
+	/* Minimum acceptable LED blinking time period (in ms) */
+	u32_t min_period;
+	/* Maximum acceptable LED blinking time period (in ms) */
+	u32_t max_period;
+	/* Minimum acceptable LED brightness value */
+	u8_t min_brightness;
+	/* Maximum acceptable LED brightness value */
+	u8_t max_brightness;
+};
+
+/**
+ * @typedef led_api_blink()
+ * @brief Callback API for blinking an LED
+ *
+ * @see led_blink() for argument descriptions.
+ */
+typedef int (*led_api_blink)(struct device *dev, u32_t led, u32_t period);
+
+/**
+ * @typedef led_api_set_brightness()
+ * @brief Callback API for setting brightness of an LED
```

Comment:
What's the relationship between 'blink' and 'set_brightness'?  As in, if I start a LED blinking, and then call set brightness, should it continue to blink?

Reply:
At this moment, we don't put any restrictions from API. It totally depends on how HW handles it. In future if we implement software blinking,  we may decide on cancelling blink when set_brightness is called.